smap_l2_gridder/collections.py

```diff
@@ -0,0 +1,124 @@
+"""Collection information.
+
+Defines hierarchies necessary for gridding collections with the SMAP L2 Gridder.
+
+While the input files are hierarchical and in hdf format, they are not fully
+self describing. Each griddable variable is tied to two variables that contain
+the column and row of a grid into which the value should be placed. But these
+column and row indices are not always in the same location.  Additionally,
+there are a number of different resolution grids and coordinate reference
+systems, that here are identified by a gpd file and epsg code that are
+described in the documentation for the dataset but are hard/impossible to
+determine by looking at the files themselves, for this reason we explictly lay
+out this information in this file along with helper routines to ease access.
+
+"""
+
+from .exceptions import InvalidCollectionError
+
+STANDARD_LOCATIONS = {
+    'row': 'Soil_Moisture_Retrieval_Data/EASE_row_index',
+    'col': 'Soil_Moisture_Retrieval_Data/EASE_column_index',
+}
+
+GRIDS = {
+    'M03km': {'gpd': 'EASE2_M03km.gpd', 'epsg': 'EPSG:6933'},
+    'M09km': {'gpd': 'EASE2_M09km.gpd', 'epsg': 'EPSG:6933'},
+    'M36km': {'gpd': 'EASE2_M36km.gpd', 'epsg': 'EPSG:6933'},
+    'N09km': {'gpd': 'EASE2_N09km.gpd', 'epsg': 'EPSG:6931'},
+}
+
+
+COLLECTION_INFORMATION = {
+    'SPL2SMP_E': {
+        'metadata': ['Metadata'],
+        'data_groups': {
+            'Soil_Moisture_Retrieval_Data': {
+                **STANDARD_LOCATIONS,
+                **GRIDS['M09km'],
+            },
+            'Soil_Moisture_Retrieval_Data_Polar': {
+                'row': 'Soil_Moisture_Retrieval_Data_Polar/EASE_row_index',
+                'col': 'Soil_Moisture_Retrieval_Data_Polar/EASE_column_index',
+                **GRIDS['N09km'],
+            },
+        },
+    },
+    'SPL2SMAP': {
+        'metadata': ['Metadata'],
+        'data_groups': {
+            'Soil_Moisture_Retrieval_Data': {
+                **STANDARD_LOCATIONS,
+                **GRIDS['M09km'],
+                'dropped_variables': ['spacecraft_overpass_time_utc'],
+            },
+            'Soil_Moisture_Retrieval_Data_3km': {
+                'row': 'Soil_Moisture_Retrieval_Data_3km/EASE_row_index_3km',
+                'col': 'Soil_Moisture_Retrieval_Data_3km/EASE_column_index_3km',
+                **GRIDS['M03km'],
+                'dropped_variables': ['spacecraft_overpass_time_utc'],
```

Comment:
The on prem system doesn't include this variable in it's outputs.  I found that including it, adds about 20x the size of the file without it. 

```
       168897423 Jan 24 14:58 output-without-timevar_SMAP_L2_SM_A_02289_D_20150707T020806_R13080_001.nc
      3349625663 Jan 23 17:24 output_SMAP_L2_SM_A_02289_D_20150707T020806_R13080_001.nc

```